Reapply "[Response Ops][Alerting] Backfill actions schema changes for intermediate release

ymao1 opened the pull request:
## Summary

Redoing [this PR](https://github.com/elastic/kibana/pull/203184) which had to be [reverted](https://github.com/elastic/kibana/pull/204218). This should not be merged until [this update to the task manager v1 schema](https://github.com/elastic/kibana/pull/204413) is released.

## To verify
1. Set `xpack.task_manager.unsafe.exclude_task_types: ['ad_hoc_run-backfill', 'actions:*']` in your Kibana config. 
2. Run Kibana on main and create some detection rules that run frequently, with actions.
3. Schedule a manual run for your detection rules.
    - Because of the config, the `action_task_params` SO and the `ad_hoc_run_task_params` SO will get written but not read yet.
4. Remove the `exclude_task_types` config and "upgrade" to this PR branch and verify that rules continue to run and that the actions are triggered and the manual rule runs go through
5. Re-add the `exclude_task_types` config and let the rule run again to schedule action. Schedule another manual rule run.
6. Remove the `exclude_task_types` config and "downgrade" back to main and verify that rules continue to run, the action gets triggered and manual rule runs go through.


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/261591)
* Commit: 5dd56fdda90bd01bc99895d41e444b8b3096c6c1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261591","number":261591,"commit":"5dd56fdda90bd01bc99895d41e444b8b3096c6c1"}],"number":261591}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/263685)
* Commit: 7d38a846761fa4f0f325ca4b58924a90eec3c3d8

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #261591](https://buildkite.com/elastic/kibana-pull-request/builds/261591) succeeded 5dd56fdda90bd01bc99895d41e444b8b3096c6c1

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263685","number":263685,"commit":"7d38a846761fa4f0f325ca4b58924a90eec3c3d8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261591","number":261591,"commit":"5dd56fdda90bd01bc99895d41e444b8b3096c6c1"}],"number":263685}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/264330)
* Commit: 4c7615458dc90079cf1c2bcf14ae3a8f806b64ec

### Failed CI Steps
* [FTR Configs #26](https://buildkite.com/elastic/kibana-pull-request/builds/264330#01943bdf-904c-4d04-b779-f42b35c22c18)
* [FTR Configs #93](https://buildkite.com/elastic/kibana-pull-request/builds/264330#01943bdf-907c-4aec-a760-8b3e389377bd)
* [Post-Build](https://buildkite.com/elastic/kibana-pull-request/builds/264330#01943bc7-0eca-4fe1-8a4b-58a2baf76c6b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/264330#01943bdf-904c-4d04-b779-f42b35c22c18) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/264330/jobs/01943bdf-904c-4d04-b779-f42b35c22c18/artifacts/01943c0c-5525-4d20-8a26-bc276c24135f) FTR Configs #<span></span>26 / discover/security/context_awareness security root profile cell renderers host.name DataView mode should open host.name flyout
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/264330#01943bdf-907c-4aec-a760-8b3e389377bd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/264330/jobs/01943bdf-907c-4aec-a760-8b3e389377bd/artifacts/01943bf0-497e-4241-9318-06d68a3f7967) FTR Configs #<span></span>93 / serverless observability UI Dataset Quality Dataset quality summary shows poor, degraded and good count as 0 and all dataset as healthy

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ‼️ ERROR: no builds found for mergeBase sha [e04b20018a441e679848e114df68ba688976c83c]

### History
* :green_heart: [Build #263685](https://buildkite.com/elastic/kibana-pull-request/builds/263685) succeeded 7d38a846761fa4f0f325ca4b58924a90eec3c3d8
* :green_heart: [Build #261591](https://buildkite.com/elastic/kibana-pull-request/builds/261591) succeeded 5dd56fdda90bd01bc99895d41e444b8b3096c6c1

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264330","number":264330,"commit":"4c7615458dc90079cf1c2bcf14ae3a8f806b64ec"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263685","number":263685,"commit":"7d38a846761fa4f0f325ca4b58924a90eec3c3d8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261591","number":261591,"commit":"5dd56fdda90bd01bc99895d41e444b8b3096c6c1"}],"number":264330}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/264465)
* Commit: 30079e0161e0e854a2d937027c6bee95da869f59

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #264330](https://buildkite.com/elastic/kibana-pull-request/builds/264330) failed 4c7615458dc90079cf1c2bcf14ae3a8f806b64ec
* :green_heart: [Build #263685](https://buildkite.com/elastic/kibana-pull-request/builds/263685) succeeded 7d38a846761fa4f0f325ca4b58924a90eec3c3d8
* :green_heart: [Build #261591](https://buildkite.com/elastic/kibana-pull-request/builds/261591) succeeded 5dd56fdda90bd01bc99895d41e444b8b3096c6c1

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264465","number":264465,"commit":"30079e0161e0e854a2d937027c6bee95da869f59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264330","number":264330,"commit":"4c7615458dc90079cf1c2bcf14ae3a8f806b64ec"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263685","number":263685,"commit":"7d38a846761fa4f0f325ca4b58924a90eec3c3d8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261591","number":261591,"commit":"5dd56fdda90bd01bc99895d41e444b8b3096c6c1"}],"number":264465}
buildkite-pr-comment-->

pmuellr reviewed (approved):
LGTM.  Test instructions in the PR got a little more involved :-), worked as described.

ymao1 commented:
@elasticmachine merge upstream

ymao1 commented:
In addition to the verification instructions, I did the following locally.

Running serverless locally with `.kibana_alerting_cases` index that had the same version conflict issue that QA project `cc5d` had and already included the new mappings in the `action_task_param` type and the `task` type (in the `.kibana_task_manager` index)

<details>
<summary> .kibana_alerting_cases/_mappings </summary>

```
{
    ".kibana_alerting_cases_1": {
        "mappings": {
            "dynamic": "strict",
            "_meta": {
                "migrationState": {
                    "convertingDocuments": false
                },
                "docVersions": {
                    "cases-rules": "10.1.0",
                    "connector_token": "10.1.0",
                    "rules-settings": "10.1.0",
                    "cases": "10.1.0",
                    "action_task_params": "10.2.0",
                    "cases-connector-mappings": "10.0.0",
                    "maintenance-window": "10.1.0",
                    "ad_hoc_run_params": "10.2.0",
                    "api_key_pending_invalidation": "10.1.0",
                    "cases-comments": "10.0.0",
                    "cases-telemetry": "10.0.0",
                    "cases-configure": "10.0.0",
                    "alert": "10.3.0",
                    "cases-user-actions": "10.0.0",
                    "action": "10.1.0"
                },
                "mappingVersions": {
                    "cases-rules": "10.1.0",
                    "connector_token": "10.1.0",
                    "rules-settings": "10.1.0",
                    "cases": "10.1.0",
                    "action_task_params": "10.2.0",
                    "cases-connector-mappings": "10.0.0",
                    "maintenance-window": "10.1.0",
                    "ad_hoc_run_params": "10.2.0",
                    "api_key_pending_invalidation": "10.1.0",
                    "cases-comments": "10.0.0",
                    "cases-telemetry": "10.0.0",
                    "cases-configure": "10.0.0",
                    "alert": "10.3.0",
                    "cases-user-actions": "10.0.0",
                    "action": "10.1.0"
                }
            },
            "properties": {
                "action": {
                    "dynamic": "false",
                    "properties": {
                        "actionTypeId": {
                            "type": "keyword"
                        },
                        "name": {
                            "type": "text",
                            "fields": {
                                "keyword": {
                                    "type": "keyword"
                                }
                            }
                        }
                    }
                },
                "action_task_params": {
                    "dynamic": "false",
                    "properties": {
                        "apiKeyId": {
                            "type": "keyword"
                        }
                    }
                },
                "ad_hoc_run_params": {
                    "dynamic": "false",
                    "properties": {
                        "apiKeyId": {
                            "type": "keyword"
                        },
                        "createdAt": {
                            "type": "date"
                        },
                        "end": {
                            "type": "date"
                        },
                        "rule": {
                            "properties": {
                                "alertTypeId": {
                                    "type": "keyword"
                                },
                                "consumer": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "start": {
                            "type": "date"
                        }
                    }
                },
                "alert": {
                    "dynamic": "false",
                    "properties": {
                        "actions": {
                            "type": "nested",
                            "dynamic": "false",
                            "properties": {
                                "actionRef": {
                                    "type": "keyword"
                                },
                                "actionTypeId": {
                                    "type": "keyword"
                                },
                                "group": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "alertTypeId": {
                            "type": "keyword"
                        },
                        "consumer": {
                            "type": "keyword"
                        },
                        "createdAt": {
                            "type": "date"
                        },
                        "createdBy": {
                            "type": "keyword"
                        },
                        "enabled": {
                            "type": "boolean"
                        },
                        "executionStatus": {
                            "properties": {
                                "error": {
                                    "properties": {
                                        "message": {
                                            "type": "keyword"
                                        },
                                        "reason": {
                                            "type": "keyword"
                                        }
                                    }
                                },
                                "lastDuration": {
                                    "type": "long"
                                },
                                "lastExecutionDate": {
                                    "type": "date"
                                },
                                "numberOfTriggeredActions": {
                                    "type": "long"
                                },
                                "status": {
                                    "type": "keyword"
                                },
                                "warning": {
                                    "properties": {
                                        "message": {
                                            "type": "keyword"
                                        },
                                        "reason": {
                                            "type": "keyword"
                                        }
                                    }
                                }
                            }
                        },
                        "lastRun": {
                            "properties": {
                                "alertsCount": {
                                    "properties": {
                                        "active": {
                                            "type": "float"
                                        },
                                        "ignored": {
                                            "type": "float"
                                        },
                                        "new": {
                                            "type": "float"
                                        },
                                        "recovered": {
                                            "type": "float"
                                        }
                                    }
                                },
                                "outcome": {
                                    "type": "keyword"
                                },
                                "outcomeOrder": {
                                    "type": "float"
                                }
                            }
                        },
                        "legacyId": {
                            "type": "keyword"
                        },
                        "mapped_params": {
                            "properties": {
                                "risk_score": {
                                    "type": "float"
                                },
                                "severity": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "monitoring": {
                            "properties": {
                                "run": {
                                    "properties": {
                                        "calculated_metrics": {
                                            "properties": {
                                                "p50": {
                                                    "type": "long"
                                                },
                                                "p95": {
                                                    "type": "long"
                                                },
                                                "p99": {
                                                    "type": "long"
                                                },
                                                "success_ratio": {
                                                    "type": "float"
                                                }
                                            }
                                        },
                                        "last_run": {
                                            "properties": {
                                                "metrics": {
                                                    "properties": {
                                                        "duration": {
                                                            "type": "long"
                                                        },
                                                        "gap_duration_s": {
                                                            "type": "float"
                                                        },
                                                        "total_alerts_created": {
                                                            "type": "float"
                                                        },
                                                        "total_alerts_detected": {
                                                            "type": "float"
                                                        },
                                                        "total_indexing_duration_ms": {
                                                            "type": "long"
                                                        },
                                                        "total_search_duration_ms": {
                                                            "type": "long"
                                                        }
                                                    }
                                                },
                                                "timestamp": {
                                                    "type": "date"
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        },
                        "muteAll": {
                            "type": "boolean"
                        },
                        "mutedInstanceIds": {
                            "type": "keyword"
                        },
                        "name": {
                            "type": "text",
                            "fields": {
                                "keyword": {
                                    "type": "keyword",
                                    "normalizer": "lowercase"
                                }
                            }
                        },
                        "notifyWhen": {
                            "type": "keyword"
                        },
                        "params": {
                            "type": "flattened",
                            "ignore_above": 4096
                        },
                        "revision": {
                            "type": "long"
                        },
                        "running": {
                            "type": "boolean"
                        },
                        "schedule": {
                            "properties": {
                                "interval": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "scheduledTaskId": {
                            "type": "keyword"
                        },
                        "snoozeSchedule": {
                            "type": "nested",
                            "properties": {
                                "duration": {
                                    "type": "long"
                                },
                                "id": {
                                    "type": "keyword"
                                },
                                "skipRecurrences": {
                                    "type": "date",
                                    "format": "strict_date_time"
                                }
                            }
                        },
                        "tags": {
                            "type": "keyword"
                        },
                        "throttle": {
                            "type": "keyword"
                        },
                        "updatedAt": {
                            "type": "date"
                        },
                        "updatedBy": {
                            "type": "keyword"
                        }
                    }
                },
                "api_key_pending_invalidation": {
                    "properties": {
                        "apiKeyId": {
                            "type": "keyword"
                        },
                        "createdAt": {
                            "type": "date"
                        }
                    }
                },
                "cases": {
                    "dynamic": "false",
                    "properties": {
                        "assignees": {
                            "properties": {
                                "uid": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "category": {
                            "type": "keyword"
                        },
                        "closed_at": {
                            "type": "date"
                        },
                        "closed_by": {
                            "properties": {
                                "email": {
                                    "type": "keyword"
                                },
                                "full_name": {
                                    "type": "keyword"
                                },
                                "profile_uid": {
                                    "type": "keyword"
                                },
                                "username": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "connector": {
                            "properties": {
                                "fields": {
                                    "properties": {
                                        "key": {
                                            "type": "text"
                                        },
                                        "value": {
                                            "type": "text"
                                        }
                                    }
                                },
                                "name": {
                                    "type": "text"
                                },
                                "type": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "created_at": {
                            "type": "date"
                        },
                        "created_by": {
                            "properties": {
                                "email": {
                                    "type": "keyword"
                                },
                                "full_name": {
                                    "type": "keyword"
                                },
                                "profile_uid": {
                                    "type": "keyword"
                                },
                                "username": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "customFields": {
                            "type": "nested",
                            "properties": {
                                "key": {
                                    "type": "keyword"
                                },
                                "type": {
                                    "type": "keyword"
                                },
                                "value": {
                                    "type": "keyword",
                                    "fields": {
                                        "boolean": {
                                            "type": "boolean",
                                            "ignore_malformed": true
                                        },
                                        "date": {
                                            "type": "date",
                                            "ignore_malformed": true
                                        },
                                        "ip": {
                                            "type": "ip",
                                            "ignore_malformed": true
                                        },
                                        "number": {
                                            "type": "long",
                                            "ignore_malformed": true
                                        },
                                        "string": {
                                            "type": "text"
                                        }
                                    }
                                }
                            }
                        },
                        "description": {
                            "type": "text"
                        },
                        "duration": {
                            "type": "unsigned_long"
                        },
                        "external_service": {
                            "properties": {
                                "connector_name": {
                                    "type": "keyword"
                                },
                                "external_id": {
                                    "type": "keyword"
                                },
                                "external_title": {
                                    "type": "text"
                                },
                                "external_url": {
                                    "type": "text"
                                },
                                "pushed_at": {
                                    "type": "date"
                                },
                                "pushed_by": {
                                    "properties": {
                                        "email": {
                                            "type": "keyword"
                                        },
                                        "full_name": {
                                            "type": "keyword"
                                        },
                                        "profile_uid": {
                                            "type": "keyword"
                                        },
                                        "username": {
                                            "type": "keyword"
                                        }
                                    }
                                }
                            }
                        },
                        "owner": {
                            "type": "keyword"
                        },
                        "settings": {
                            "properties": {
                                "syncAlerts": {
                                    "type": "boolean"
                                }
                            }
                        },
                        "severity": {
                            "type": "short"
                        },
                        "status": {
                            "type": "short"
                        },
                        "tags": {
                            "type": "keyword"
                        },
                        "title": {
                            "type": "text",
                            "fields": {
                                "keyword": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "total_alerts": {
                            "type": "integer"
                        },
                        "total_comments": {
                            "type": "integer"
                        },
                        "updated_at": {
                            "type": "date"
                        },
                        "updated_by": {
                            "properties": {
                                "email": {
                                    "type": "keyword"
                                },
                                "full_name": {
                                    "type": "keyword"
                                },
                                "profile_uid": {
                                    "type": "keyword"
                                },
                                "username": {
                                    "type": "keyword"
                                }
                            }
                        }
                    }
                },
                "cases-comments": {
                    "dynamic": "false",
                    "properties": {
                        "actions": {
                            "properties": {
                                "type": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "alertId": {
                            "type": "keyword"
                        },
                        "comment": {
                            "type": "text"
                        },
                        "created_at": {
                            "type": "date"
                        },
                        "created_by": {
                            "properties": {
                                "username": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "externalReferenceAttachmentTypeId": {
                            "type": "keyword"
                        },
                        "owner": {
                            "type": "keyword"
                        },
                        "persistableStateAttachmentTypeId": {
                            "type": "keyword"
                        },
                        "pushed_at": {
                            "type": "date"
                        },
                        "type": {
                            "type": "keyword"
                        },
                        "updated_at": {
                            "type": "date"
                        }
                    }
                },
                "cases-configure": {
                    "dynamic": "false",
                    "properties": {
                        "closure_type": {
                            "type": "keyword"
                        },
                        "created_at": {
                            "type": "date"
                        },
                        "owner": {
                            "type": "keyword"
                        }
                    }
                },
                "cases-connector-mappings": {
                    "dynamic": "false",
                    "properties": {
                        "owner": {
                            "type": "keyword"
                        }
                    }
                },
                "cases-rules": {
                    "dynamic": "false",
                    "properties": {
                        "counter": {
                            "type": "unsigned_long"
                        },
                        "createdAt": {
                            "type": "date"
                        },
                        "rules": {
                            "properties": {
                                "id": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "updatedAt": {
                            "type": "date"
                        }
                    }
                },
                "cases-telemetry": {
                    "type": "object",
                    "dynamic": "false"
                },
                "cases-user-actions": {
                    "dynamic": "false",
                    "properties": {
                        "action": {
                            "type": "keyword"
                        },
                        "created_at": {
                            "type": "date"
                        },
                        "created_by": {
                            "properties": {
                                "username": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "owner": {
                            "type": "keyword"
                        },
                        "payload": {
                            "dynamic": "false",
                            "properties": {
                                "assignees": {
                                    "properties": {
                                        "uid": {
                                            "type": "keyword"
                                        }
                                    }
                                },
                                "comment": {
                                    "properties": {
                                        "externalReferenceAttachmentTypeId": {
                                            "type": "keyword"
                                        },
                                        "persistableStateAttachmentTypeId": {
                                            "type": "keyword"
                                        },
                                        "type": {
                                            "type": "keyword"
                                        }
                                    }
                                },
                                "connector": {
                                    "properties": {
                                        "type": {
                                            "type": "keyword"
                                        }
                                    }
                                }
                            }
                        },
                        "type": {
                            "type": "keyword"
                        }
                    }
                },
                "connector_token": {
                    "dynamic": "false",
                    "properties": {
                        "connectorId": {
                            "type": "keyword"
                        },
                        "tokenType": {
                            "type": "keyword"
                        }
                    }
                },
                "coreMigrationVersion": {
                    "type": "keyword"
                },
                "created_at": {
                    "type": "date"
                },
                "created_by": {
                    "type": "keyword"
                },
                "maintenance-window": {
                    "dynamic": "false",
                    "properties": {
                        "enabled": {
                            "type": "boolean"
                        },
                        "events": {
                            "type": "date_range",
                            "format": "epoch_millis||strict_date_optional_time"
                        }
                    }
                },
                "managed": {
                    "type": "boolean"
                },
                "namespace": {
                    "type": "keyword"
                },
                "namespaces": {
                    "type": "keyword"
                },
                "originId": {
                    "type": "keyword"
                },
                "references": {
                    "type": "nested",
                    "properties": {
                        "id": {
                            "type": "keyword"
                        },
                        "name": {
                            "type": "keyword"
                        },
                        "type": {
                            "type": "keyword"
                        }
                    }
                },
                "rules-settings": {
                    "dynamic": "false",
                    "properties": {
                        "flapping": {
                            "type": "object"
                        }
                    }
                },
                "type": {
                    "type": "keyword"
                },
                "typeMigrationVersion": {
                    "type": "version"
                },
                "updated_at": {
                    "type": "date"
                },
                "updated_by": {
                    "type": "keyword"
                }
            }
        }
    }
}
```
</details>

<details>
<summary> .kibana_task_manager/_mappings </summary>

```
{
    ".kibana_task_manager_1": {
        "mappings": {
            "dynamic": "strict",
            "_meta": {
                "migrationState": {
                    "convertingDocuments": false
                },
                "docVersions": {
                    "task": "10.3.0",
                    "background-task-node": "10.1.0"
                },
                "mappingVersions": {
                    "task": "10.3.0",
                    "background-task-node": "10.1.0"
                }
            },
            "properties": {
                "background-task-node": {
                    "dynamic": "false",
                    "properties": {
                        "id": {
                            "type": "keyword"
                        },
                        "last_seen": {
                            "type": "date"
                        }
                    }
                },
                "coreMigrationVersion": {
                    "type": "keyword"
                },
                "created_at": {
                    "type": "date"
                },
                "created_by": {
                    "type": "keyword"
                },
                "managed": {
                    "type": "boolean"
                },
                "namespace": {
                    "type": "keyword"
                },
                "namespaces": {
                    "type": "keyword"
                },
                "originId": {
                    "type": "keyword"
                },
                "references": {
                    "type": "nested",
                    "properties": {
                        "id": {
                            "type": "keyword"
                        },
                        "name": {
                            "type": "keyword"
                        },
                        "type": {
                            "type": "keyword"
                        }
                    }
                },
                "task": {
                    "dynamic": "false",
                    "properties": {
                        "attempts": {
                            "type": "integer"
                        },
                        "enabled": {
                            "type": "boolean"
                        },
                        "ownerId": {
                            "type": "keyword"
                        },
                        "partition": {
                            "type": "integer"
                        },
                        "priority": {
                            "type": "integer"
                        },
                        "retryAt": {
                            "type": "date"
                        },
                        "runAt": {
                            "type": "date"
                        },
                        "schedule": {
                            "properties": {
                                "interval": {
                                    "type": "keyword"
                                }
                            }
                        },
                        "scheduledAt": {
                            "type": "date"
                        },
                        "scope": {
                            "type": "keyword"
                        },
                        "status": {
                            "type": "keyword"
                        },
                        "taskType": {
                            "type": "keyword"
                        }
                    }
                },
                "type": {
                    "type": "keyword"
                },
                "typeMigrationVersion": {
                    "type": "version"
                },
                "updated_at": {
                    "type": "date"
                },
                "updated_by": {
                    "type": "keyword"
                }
            }
        }
    }
}
```
</details>

I did the following:

1. Started up Kibana, observed the mapping conflict error.
2. Updated the `mappingVersions` for `.kibana_alerting_cases`, started up Kibana with just the migrator node, observed the migration error.
3. Updated the `docVersions` for `.kibana_alerting_cases`, started up Kibana with just the migrator node, startup 👍, started up Kibana with [bg,ui] roles, startup 👍 
4. Switched to this branch, started up Kibana with just the migrator node, startup 👍, started up Kibana with [bg,ui] roles, startup 👍 
5. After this "upgrade", I see the `docVersion` and `mappingVersion` for `action_task_params` and `ad_hoc_run_params` updated back to 10.2.0

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/264929)
* Commit: 5d67b9b059a1e5c60a991346814c57af4bd60784

### Failed CI Steps
* [Serverless Rule Management - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/264929#01944208-9c54-46f4-ad92-beaea9d89b4c)

### History
* :green_heart: [Build #264465](https://buildkite.com/elastic/kibana-pull-request/builds/264465) succeeded 30079e0161e0e854a2d937027c6bee95da869f59
* :broken_heart: [Build #264330](https://buildkite.com/elastic/kibana-pull-request/builds/264330) failed 4c7615458dc90079cf1c2bcf14ae3a8f806b64ec
* :green_heart: [Build #263685](https://buildkite.com/elastic/kibana-pull-request/builds/263685) succeeded 7d38a846761fa4f0f325ca4b58924a90eec3c3d8
* :green_heart: [Build #261591](https://buildkite.com/elastic/kibana-pull-request/builds/261591) succeeded 5dd56fdda90bd01bc99895d41e444b8b3096c6c1

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264929","number":264929,"commit":"5d67b9b059a1e5c60a991346814c57af4bd60784"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264465","number":264465,"commit":"30079e0161e0e854a2d937027c6bee95da869f59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264330","number":264330,"commit":"4c7615458dc90079cf1c2bcf14ae3a8f806b64ec"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263685","number":263685,"commit":"7d38a846761fa4f0f325ca4b58924a90eec3c3d8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261591","number":261591,"commit":"5dd56fdda90bd01bc99895d41e444b8b3096c6c1"}],"number":264929}
buildkite-pr-comment-->

ymao1 commented:
@elasticmachine merge upstream

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/264946)
* Commit: 8ccdb64abe5f0dfd8090cfe05bd65216457caaf9

### Failed CI Steps
* [FTR Configs #3](https://buildkite.com/elastic/kibana-pull-request/builds/264946#01944279-f85d-4bc3-971a-479a63fa3e4d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/264946#01944279-f85d-4bc3-971a-479a63fa3e4d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/264946/jobs/01944279-f85d-4bc3-971a-479a63fa3e4d/artifacts/0194429b-3145-4122-aec9-055c1e4f460c) FTR Configs #<span></span>3 / Rules Management - Prebuilt Rules - Prebuilt Rules Management @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Bootstrap Prebuilt Rules should skip installing fleet packages if they are already installed

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #264929](https://buildkite.com/elastic/kibana-pull-request/builds/264929) failed 5d67b9b059a1e5c60a991346814c57af4bd60784
* :green_heart: [Build #264465](https://buildkite.com/elastic/kibana-pull-request/builds/264465) succeeded 30079e0161e0e854a2d937027c6bee95da869f59
* :broken_heart: [Build #264330](https://buildkite.com/elastic/kibana-pull-request/builds/264330) failed 4c7615458dc90079cf1c2bcf14ae3a8f806b64ec
* :green_heart: [Build #263685](https://buildkite.com/elastic/kibana-pull-request/builds/263685) succeeded 7d38a846761fa4f0f325ca4b58924a90eec3c3d8
* :green_heart: [Build #261591](https://buildkite.com/elastic/kibana-pull-request/builds/261591) succeeded 5dd56fdda90bd01bc99895d41e444b8b3096c6c1

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264946","number":264946,"commit":"8ccdb64abe5f0dfd8090cfe05bd65216457caaf9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264929","number":264929,"commit":"5d67b9b059a1e5c60a991346814c57af4bd60784"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264465","number":264465,"commit":"30079e0161e0e854a2d937027c6bee95da869f59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/264330","number":264330,"commit":"4c7615458dc90079cf1c2bcf14ae3a8f806b64ec"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/263685","number":263685,"commit":"7d38a846761fa4f0f325ca4b58924a90eec3c3d8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/261591","number":261591,"commit":"5dd56fdda90bd01bc99895d41e444b8b3096c6c1"}],"number":264946}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/205821">](https://github.com/elastic/kibana/pull/205821)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
